### ISSUE TITLE
dataclass_transform ignores kw_only argument in class definition (Metaclass Keyword Argument)

### ISSUE THREAD

**Issue by arjenzorgdoc** [astral-sh/ty#3033]
### Summary

```python
from dataclasses import dataclass
from typing import dataclass_transform, Any


@dataclass_transform(kw_only_default=False)
class MyBase:
    def __init_subclass__(cls, kw_only: bool = False, **kwargs: Any) -> None:
        pass


class Foo1(MyBase, kw_only=True):
    name: str = "john"
    age: int   # False positive: dataclass-field-order

Foo1("a", 2), # False negative
```

https://play.ty.dev/bb345be7-ea4d-446c-b123-600132cef2ee

**Comment by sharkdp** [comment]
Thank you for reporting this. That is a bug in ty.
